core/staking_verifier.go

```diff
@@ -264,7 +267,13 @@ func VerifyAndUndelegateFromMsg(
 			if err := wrapper.SanityCheck(
 				staking.DoNotEnforceMaxBLS,
 			); err != nil {
-				return nil, err
+				// allow self delegation to go below min self delegation
+				// but set the status to inactive
+				if strings.Contains(err.Error(), invalidSelfDelMsg) {
```

Comment:
I don't like this. Why are we checking an error string value? Never depend on the format/impl detail of the error message itself.